src/app/external-cluster-wizard/steps/external-cluster/provider/aks/component.ts

```diff
@@ -253,19 +261,36 @@ export class AKSClusterSettingsComponent
         this.vmSizeLabel = this.vmSizes?.length ? VMSizeState.Ready : VMSizeState.Empty;
         this.control(Controls.VmSize).enable();
       });
-      this._getAKSAvailableNodePoolVersionsForCreateMachineDeployment().subscribe(
-        (nodePoolVersions: AKSNodePoolVersionForMachineDeployments[]) => {
-          this.nodePoolVersionsForMD = nodePoolVersions.map(nodePoolVersion => nodePoolVersion.version);
-        }
-      );
+
+      this.nodePoolVersionsForMDLabel = NodePoolVersionState.Loading;
+      this._getAKSAvailableNodePoolVersionsForCreateMachineDeployment()
+        .pipe(tap(_ => this._clearNodePoolVersions()))
+        .subscribe((nodePoolVersions: AKSNodePoolVersionForMachineDeployments[]) => {
+          this.nodePoolVersionsForMD = nodePoolVersions;
+          this.nodePoolVersionsForMDLabel = this.nodePoolVersionsForMD?.length
+            ? NodePoolVersionState.Ready
+            : NodePoolVersionState.Empty;
+          nodePoolVersions.forEach((nodePoolVersion: AKSNodePoolVersionForMachineDeployments) => {
+            if (nodePoolVersion.default) {
+              this.control(Controls.KubernetesVersion).setValue(nodePoolVersion.version);
+            }
+          });
+        });
     } else {
       this.control(Controls.Location)
         .valueChanges.pipe(debounceTime(this._debounceTime))
         .pipe(tap(_ => this._clearVmSize()))
         .pipe(
-          switchMap((location: string) => {
+          switchMap((location: AKSLocation) => {
+            let obs$;
```

Comment:
do we need a type here

Reply:
it's for static type checking yes `location` is no more string but a return object from combobox component